carInsuranceBackEnd/src/main/java/com/yakvel/carInsuranceBackEnd/controllers/manager/ManagerTicketController.java

```diff
@@ -0,0 +1,147 @@
+        Person manager = (Person) SecurityContextHolder.getContext().getAuthentication().getPrincipal();
+        Set<TicketDto> tickets = ticketRepository.findAll()
+                .stream()
+                .filter(ticket -> ticket.getTicketOwner().getInsuranceCompany().equals(manager.getInsuranceCompany()))
+                .map(ticket -> ticketMapper.toDto(ticket))
+                .collect(Collectors.toSet());
+        return ResponseEntity.ok(tickets);
+    }
+
+    @GetMapping("/ticket/{ticketId}")
+    public ResponseEntity getManagerTicketDetails(@PathVariable long ticketId) {
+        Person manager = (Person) SecurityContextHolder.getContext().getAuthentication().getPrincipal();
+        Ticket ticket = ticketRepository.findById(ticketId).orElse(null);
+
+        ResponseEntity<String> checkAnswer = checkTicket(manager, ticket);
+        if (checkAnswer!=null) {return checkAnswer;}
+        TicketDto ticketDto = ticketMapper.toDto(ticket);
+
+        return ResponseEntity.ok(ticketDto);
+    }
+
+    @PutMapping("/ticket/{ticketId}")
+    public ResponseEntity<String> updateTicketFromManager(@PathVariable long ticketId, @RequestBody TicketManagerDto dto) {
+        Person manager = (Person) SecurityContextHolder.getContext().getAuthentication().getPrincipal();
+        Ticket ticket = ticketRepository.findById(ticketId).orElse(null);
+
+        ResponseEntity<String> checkAnswer = checkTicket(manager, ticket);
+        if (checkAnswer!=null) {return checkAnswer;}
+
+        mergeTicket(dto, ticket,manager);
+        ticketRepository.save(ticket);
+        return ResponseEntity.ok("Ticket was updated!");
+    }
+
+    @PostMapping("/ticket/{ticketId}/comment")
+    public ResponseEntity<String> addCommentToTicket(@PathVariable long ticketId, @RequestBody CommentDto dto) {
+        Person manager = (Person) SecurityContextHolder.getContext().getAuthentication().getPrincipal();
+        Ticket ticket = ticketRepository.findById(ticketId).orElse(null);
+
+        ResponseEntity<String> checkAnswer = checkTicket(manager, ticket);
+        if (checkAnswer!=null) {return checkAnswer;}
+
+        Comment comment = prepareComment(dto, manager, ticket);
+        commentRepository.save(comment);
+        return ResponseEntity.ok("Comment was added!");
+    }
+    @DeleteMapping("/ticket/comment/{commentId}")
+    public ResponseEntity<String> deleteCommentToTicket(@PathVariable long commentId) {
+        Person manager = (Person) SecurityContextHolder.getContext().getAuthentication().getPrincipal();
+        Comment comment = commentRepository.findById(commentId).orElse(null);
+
+        ResponseEntity<String> checkAnswer = checkComment(manager, comment);
+        if (checkAnswer!=null) {return checkAnswer;}
+
+        commentRepository.delete(comment);
+        return ResponseEntity.ok("Comment was deleted!");
+    }
+
+    private static ResponseEntity<String> checkTicket(Person manager, Ticket ticket) {
```

Comment:
the name of the method is not very descriptive. I would suggest renaming it.
Also this is a good example when Optional can be used instead of null return values
```suggestion
    private static Optional<ResponseEntity<String>> checkTicketOwnership(Person manager, Ticket ticket) {
```

Reply:
I thought about it , but it returns 2 variants of Response entity and we have 1 variant if ticket doesn't exist , so I desided keep it checkTicket) Okay I will change it

Reply:
done